### PR TITLE
feat: ts updates

### PR DESCRIPTION
**Pull request by tmm**
## Description

- Bumps TypeScript to 4.9 (and few other dev dependencies)
- Uses [`satisfies`](https://devblogs.microsoft.com/typescript/announcing-typescript-4-9/#the-satisfies-operator) operator for `Chain`
- Adds `Window['Ethereum']` type

## Additional Information

- [x] I read the [contributing docs](/wagmi-dev/references/blob/main/.github/CONTRIBUTING.md) (if this is your first contribution)

Your ENS/address: awkweb.eth

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -53,5 +53,10 @@
   "repository": {
     "type": "git",
     "url": "https://github.com/wagmi-dev/references.git"
+  },
+  "pnpm": {
+    "overrides": {
+      "esbuild": "^0.15.13"
+    }
   }
```
Comment (tmm) [inline package.json:61]: esbuild started supporting the `satisfies` operator in [`0.5.13`](https://github.com/evanw/esbuild/releases/tag/v0.15.13), but `tsup` is set to [`0.5.11`](https://github.com/egoist/tsup/blob/dev/package.json#L34)

File: packages/connectors/tsup.config.ts [hunk 2]
```diff
@@ -8,12 +8,12 @@ export default defineConfig(
...
       'src/index.ts',
-      'src/injected.ts',
       'src/coinbaseWallet.ts',
+      'src/injected.ts',
+      'src/ledger.ts',
       'src/metaMask.ts',
       'src/mock/index.ts',
       'src/walletConnect.ts',
-      'src/ledger.ts',
     ],
```
Comment (tmm) [inline packages/connectors/tsup.config.ts:17]: Alphabetized so output (e.g. `package.json#exports`) is also.

File: scripts/tsup.ts [hunk 3]
```diff
@@ -172,8 +173,21 @@ async function generateProxyPackages(exports: Exports) {
...
+    files.add(`/${file}`)
   }
 
+  files.add('/dist')
+  const packageJson = await fs.readJSON('package.json')
+  packageJson.files = [...files.values()]
+  await fs.writeFile(
+    'package.json',
+    JSON.stringify(packageJson, null, 2) + '\n',
+  )
```
Comment (tmm) [inline scripts/tsup.ts:188]: Adds proxy package directories to `package.json#files`.

File: packages/connectors/src/injected.ts [hunk 4]
```diff
@@ -26,7 +27,7 @@ export type InjectedConnectorOptions = {
    * @default
    * () => typeof window !== 'undefined' ? window.ethereum : undefined
    */
-  getProvider?: () => Window['ethereum'] | undefined
+  getProvider?: () => Ethereum | undefined
```
Comment (tmm) [inline packages/connectors/src/injected.ts:30]: Think it makes more sense to use these type explicitly instead of relying on `Window` type augementation.

**Review by tmm — COMMENTED** [review]
Added `Ethereum` type here since it seems like more of an implementation detail of the `InjectedConnector` than a standalone type that should live inside `@wagmi/core` (it's exported so we can bring into `@wagmi/core` once this is merged in). Will make it easier when folks need to update `InjectedProviderFlags`, etc. too.

File: .changeset/early-olives-wink.md [hunk 5]
```diff
@@ -0,0 +1,6 @@
+---
+'@wagmi/connectors': patch
```
Comment (jxom) [inline .changeset/early-olives-wink.md:2]: Wonder if this is okay as a patch. Technically is a breaking change as we are updating TypeScript, right?

Comment (tmm) [reply .changeset/early-olives-wink.md:2]: Good point! Will switch this up to `minor` (and make sure downstream also follows that).

**Review by jxom — APPROVED** [review]

**Comment by tmm** [comment]
@SocketSecurity ignore esbuild@0.15.13 esbuild@0.16.17

**Pull request merged**